app/models/report.rb

```diff
@@ -1,14 +1,16 @@
 # frozen_string_literal: true
 
 class Report
-  attr_reader :filter_params
+  attr_reader :user, :body_part, :timeframe
   TIMEFRAMES = [["Past Week", 7],
     ["Past Two Weeks", 14],
     ["Past Month", 30],
     ["Past Year", 360]].freeze
 
-  def initialize(filter_params)
-    @filter_params = filter_params
+  def initialize(user:, timeframe:, body_part_id:)
+    @user = user
+    @timeframe = timeframe.presence&.to_i
+    @body_part = BodyPart.find_by(id: body_part_id)
   end
```

Comment:
The `filter_params` hash was cumbersome. Adding the params as their own attributes follows object convention more clearly and also gives us access to the data more easily.